mbus-api/src/lib.rs

```diff
@@ -0,0 +1,196 @@
+    /// Reply to requested Channel
+    Reply(String),
+}
+
+impl FromStr for Channel {
+    type Err = String;
+
+    fn from_str(source: &str) -> Result<Self, Self::Err> {
+        match source {
+            "default" => Ok(Self::Default),
+            "registry" => Ok(Self::Registry),
+            "kiiss" => Ok(Self::Kiiss),
+            _ => Err(format!("Could not parse the channel: {}", source)),
+        }
+    }
+}
+
+impl Default for Channel {
+    fn default() -> Self {
+        Channel::Default
+    }
+}
+
+impl std::fmt::Display for Channel {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        match self {
+            Channel::Default => write!(f, "default"),
+            Channel::Registry => write!(f, "registry"),
+            Channel::Kiiss => write!(f, "kiiss"),
+            Channel::Reply(ch) => write!(f, "{}", ch),
+        }
+    }
+}
+
+/// Message id which uniquely identifies every type of unsolicited message
+/// The solicited (replies) message do not currently carry an id as they
+/// are sent to a specific requested channel
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+pub enum MessageId {
+    /// Default
+    Default,
+    /// Update Config
+    UpdateConfig,
+    /// Request Config
+    RequestConfig,
+    /// Get Svc Name Dummy
+    GetSvcName,
+    /// Register mayastor
+    Register,
+    /// Deregister mayastor
+    Deregister,
+}
+
+/// Sender identification (eg which mayastor instance sent the message)
+pub type SenderId = String;
+
+/// Mayastor configurations
+/// Currently, we have the global mayastor config and the child states config
+#[derive(
+    Serialize, Deserialize, Debug, Clone, Eq, PartialEq, std::hash::Hash,
+)]
+pub enum ConfigKind {
+    /// Mayastor global config
+    MayastorConfig,
+    /// Mayastor child states config
+    ChildStatesConfig,
+}
+impl Default for ConfigKind {
+    fn default() -> Self {
+        ConfigKind::MayastorConfig
+    }
+}
+
+/// Config Messages
+
+/// Update mayastor configuration
+#[derive(Serialize, Deserialize, Debug, Default, Clone)]
+pub struct UpdateConfig {
+    /// type of config being updated
+    pub kind: ConfigKind,
+    /// actual config data
+    pub data: Vec<u8>,
+}
+bus_impl_all!(UpdateConfig, ());
+
+/// Request message configuration used by mayastor to request configuration
+/// from a control plane service
+#[derive(Serialize, Deserialize, Debug, Default, Clone)]
+pub struct RequestConfig {
+    /// type of config requested
+    pub kind: ConfigKind,
+}
+/// Reply message configuration returned by a controle plane service to mayastor
+#[derive(Serialize, Deserialize, Debug, Default, Clone)]
+pub struct ReplyConfig {
+    /// config data
+    pub config: Vec<u8>,
+}
+bus_impl_all!(GetConfig, RequestConfig, ReplyConfig);
+
+/// Registration
+
+/// Register message payload
+#[derive(Serialize, Deserialize, Default, Debug, Clone)]
+pub struct Register {
+    /// id of the mayastor instance
+    pub id: String,
```

Comment:
These were just moved out from the mayastor into the lib.
I think the id is a string because that's how mayastor names the mayastor instances, though I guess we could type alias it.